### PR TITLE
Document constraints.txt usage.

### PR DESCRIPTION
**Pull request by loechel**
I have documented the corresponding constrains.txt usage with tox, as it work similar to requirements.txt but was not documented.

Here's a quick checklist of what we'd like you to think off:
- [x] Add yourself to `CONTRIBUTORS`;
- [x] make a descriptive Pull Request text (it will be used for changelog)

### CODE REVIEW

File: doc/example/basic.txt [hunk 1]
```diff
@@ -97,6 +97,18 @@ All installation commands are executed using ``{toxinidir}``
 working directory.  Therefore, the underlying ``pip`` installation
 will assume ``requirements.txt`` to exist at ``{toxinidir}/requirements.txt``.
 
+defining constrains
+-----------------------------------------------
+
+.. versionadded:: 1.6.1
+
+(experimental) If you have a ``constrains.txt`` file you can add it to your ``deps``varaiable like this::
```
Comment (obestwalter) [inline doc/example/basic.txt]: typo: varaiable

File: doc/example/basic.txt [hunk 2]
```diff
@@ -97,6 +97,18 @@ All installation commands are executed using ``{toxinidir}``
...
+
+    deps = constrains.txt
```
Comment (obestwalter) [inline doc/example/basic.txt]: This would have to be `deps = -cconstrains.txt`

**Comment by obestwalter** [comment]
Hi @loechel thanks for the PR. I guess you mean this: https://pip.pypa.io/en/stable/user_guide/#constraints-files?

If yes: this is nothing that is implemented as such - it is basically a side effect of the way how deps is implemented. deps passes the `-<someflag>whatever.txt` through to pip, so you theoretically use any flag that would make sense to pip there.

Maybe it would be better explain that in above section documenting `-r` that mentions that it is also possible to pass e.g. the `-c` flag with a constraints file?

File: CHANGELOG [hunk 3]
```diff
@@ -9,6 +9,7 @@ Not released yet
 - #518: Forward `USERPROFILE` by default on Windows.
 - #515: Don't require environment variables in test environments
 	where they are not used.
+- : Add documentation for constrains.txt
```
Comment (obestwalter) [inline CHANGELOG]: s/constrains/constraints/g

**Comment by loechel** [comment]
@obestwalter thanks for the feedback, I have adjusted the documentation. I hope it is better now and ready to merge.

**Pull request merged**